code/obj/machinery/plantpot.dm

```diff
@@ -1229,14 +1233,26 @@ TYPEINFO(/obj/machinery/plantpot)
 
 /// Handles the finalisation of cropcount
 /obj/machinery/plantpot/proc/HYPharvesting_finalise_cropcount(datum/HYPharvesting_data/h_data)
+	// only apply the yield multiplier and minimum if the count is likely to be positive, this allows prior modifiers to zero crop count without
+	// this overriding it back to 1.
+	var/cropsize = h_data.DNA?.get_effective_value("cropsize")
+	if (h_data.cropcount > 0)
+		h_data.cropcount *= (1 + ((cropsize * h_data.growing.yield_multi) / 100))
+		// It's too easy for low-output plants to have no cropcount at barely-negative yield values, so yield shouldn't be able to zero cropcount by itself.
+		h_data.cropcount = max(h_data.cropcount, 1)
```

Comment:
On the other hand, I think a _significantly_ negative yield should probably result in 0 produce - I think the `round(X, 1)` at the bottom should take care of any slightly negative values as well